src/Elliptic/FirstOrderOperator.hpp

```diff
@@ -43,34 +43,41 @@ struct FirstOrderFluxesImpl<Dim, tmpl::list<PrimalFields...>,
   }
 };
 
-template <typename PrimalFields, typename AuxiliaryFields,
+template <size_t Dim, typename PrimalFields, typename AuxiliaryFields,
           typename SourcesComputer>
 struct FirstOrderSourcesImpl;
 
-template <typename... PrimalFields, typename... AuxiliaryFields,
+template <size_t Dim, typename... PrimalFields, typename... AuxiliaryFields,
           typename SourcesComputer>
-struct FirstOrderSourcesImpl<tmpl::list<PrimalFields...>,
+struct FirstOrderSourcesImpl<Dim, tmpl::list<PrimalFields...>,
                              tmpl::list<AuxiliaryFields...>, SourcesComputer> {
   template <typename VarsTags, typename... SourcesArgs>
   static constexpr void apply(
       const gsl::not_null<
           Variables<db::wrap_tags_in<::Tags::Source, VarsTags>>*>
           sources,
       const Variables<VarsTags>& vars,
+      const Variables<db::wrap_tags_in<
+          ::Tags::Flux, VarsTags, tmpl::size_t<Dim>, Frame::Inertial>>& fluxes,
       const SourcesArgs&... sources_args) noexcept {
-    // Compute sources for primal fields
-    SourcesComputer::apply(
-        make_not_null(&get<::Tags::Source<PrimalFields>>(*sources))...,
-        sources_args..., get<PrimalFields>(vars)...);
-    // Compute sources for auxiliary fields. They are just the auxiliary field
-    // values.
+    // Set auxiliary field sources to the auxiliary field values to begin with.
+    // This is the standard choice, since the auxiliary equations define the
+    // auxiliary variables. Source computers can adjust or add to these sources.
     tmpl::for_each<tmpl::list<AuxiliaryFields...>>(
-        [&sources, &vars ](const auto auxiliary_field_tag_v) noexcept {
+        [&sources, &vars](const auto auxiliary_field_tag_v) noexcept {
           using auxiliary_field_tag =
               tmpl::type_from<decltype(auxiliary_field_tag_v)>;
           get<::Tags::Source<auxiliary_field_tag>>(*sources) =
               get<auxiliary_field_tag>(vars);
         });
+    // Call into the sources computer to set primal field sources and possibly
+    // adjust auxiliary field sources
+    SourcesComputer::apply(
+        make_not_null(&get<::Tags::Source<PrimalFields>>(*sources))...,
+        make_not_null(&get<::Tags::Source<AuxiliaryFields>>(*sources))...,
```

Comment:
You keep saying "instead of the auxiliary fields" but here you pass the aux fields. Please correct this

Reply:
No this passes the _sources_ for the auxiliary fields so terms can be added to them, such as Christoffel symbol terms (see this comment https://github.com/sxs-collaboration/spectre/pull/2553#discussion_r509150375). Does that make sense?

Reply:
Okay, that makes sense.

I still don't understand why you say "instead of" because the code didn't previously pass the aux vars.  Why wouldn't you need the aux vars (I realize they weren't passed before)? Is that because technically they are just (roughly) the derivatives of the primal fields? If so, I'm a bit confused that stability and well-posedness aren't adversely affected by never using the aux variables, though explaining that is beyond the scope of a GitHub comment, I think ;)

Reply:
Previously I didn't pass
(i) the auxiliary _sources_, because they are "usually" just the auxiliary vars, i.e. when no Christoffel-terms need to be added (e.g. consider the Poisson auxiliary equation `-grad(u) + v = 0` where the sources are just `v`. Remember the auxiliary equation essentially _defines_ the aux vars, which is why the aux vars are always its sources). The aux sources are set to the aux vars just a few lines above.
(ii) the auxiliary _variables_, because none of the systems that are currently implemented needs to compute non-trivial sources: Both the Poisson and the (linear) elasticity equations are only sourced by a fixed function f(x).

In the comments I say "instead of" because the sources are a function of the variables (primal and auxiliary), but _instead of_ the aux vars I pass the primal fluxes as an optimization, as we discussed before.

Does this clear things up?

Reply:
Yes, but I think this lengthy discussion suggests that you should either remove the `instead of` or expand on it in both the docs and the git message

Reply:
Adjusted the commit message and the code comment in the fixup commit, please take a look.